### PR TITLE
Use gtest

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst or @toregge PR

### CODE REVIEW

**Review by toregge — APPROVED** [review]

File: searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp [hunk 1]
```diff
@@ -64,7 +65,7 @@ struct MyCountJobRunner : public IMaintenanceJobRunner {
     void run() override { ++runCount; }
 };
 
-struct ControllerFixtureBase
+struct ControllerFixtureBase : public ::testing::Test
```
Comment (geirst) [inline searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp:68]: Consider BucketMoveJobTestBase instead.

File: searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp [hunk 2]
```diff
@@ -183,587 +184,587 @@ struct OnlyReadyControllerFixture : public ControllerFixtureBase
     }
 };
 
-TEST_F("require that nothing is moved if bucket state says so", ControllerFixture)
+TEST_F(ControllerFixture, require_that_nothing_is_moved_if_bucket_state_says_so)
```
Comment (geirst) [inline searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp:187]: Consider BucketMoveJobTest instead.

File: searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp [hunk 3]
```diff
@@ -183,587 +184,587 @@ struct OnlyReadyControllerFixture : public ControllerFixtureBase
...
+    EXPECT_FALSE(_bmj.done());
+    _bmj.scanAndMove(0, 3); // move delayed and de-activated bucket 1
+    EXPECT_TRUE(_bmj.done());
+    EXPECT_EQ(3u, docsMoved().size());
+    EXPECT_EQ(1u, bucketsModified().size());
+    EXPECT_EQ(_ready.bucket(1), bucketsModified()[0]);
 }
 
-TEST_F("require that de-activated bucket is moved before other buckets", OnlyReadyControllerFixture)
+TEST_F(OnlyReadyControllerFixture, require_that_de_activated_bucket_is_moved_before_other_buckets)
```
Comment (geirst) [inline searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp:624]: Consider OnlyReadyBucketMoveJobTest instead.

File: searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp [hunk 4]
```diff
@@ -775,55 +776,59 @@ struct ResourceLimitControllerFixture : public ControllerFixture
...
+        EXPECT_EQ(0u, bucketsModified().size());
     }
 };
 
-TEST_F("require that bucket move stops when disk limit is reached", ResourceLimitControllerFixture)
+struct ResourceLimitControllerFixture_1_2 : public ResourceLimitControllerFixture {
+    ResourceLimitControllerFixture_1_2() : ResourceLimitControllerFixture(1.2) {}
+};
+
+TEST_F(ResourceLimitControllerFixture, require_that_bucket_move_stops_when_disk_limit_is_reached)
```
Comment (geirst) [inline searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp:814]: Consider ResourceLimitTest instead.

File: searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp [hunk 5]
```diff
@@ -775,55 +776,59 @@ struct ResourceLimitControllerFixture : public ControllerFixture
...
 
-TEST_F("require that bucket move stops when memory limit is reached", ResourceLimitControllerFixture)
+TEST_F(ResourceLimitControllerFixture, require_that_bucket_move_stops_when_memory_limit_is_reached)
 {
-    f.testJobStopping(DiskMemUsageState(ResourceUsageState(), ResourceUsageState(0.7, 0.8)));
+    testJobStopping(DiskMemUsageState(ResourceUsageState(), ResourceUsageState(0.7, 0.8)));
 }
 
-TEST_F("require that bucket move uses resource limit factor for disk resource limit", ResourceLimitControllerFixture(1.2))
+TEST_F(ResourceLimitControllerFixture_1_2, require_that_bucket_move_uses_resource_limit_factor_for_disk_resource_limit)
```
Comment (geirst) [inline searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp:824]: Consider ResourceLimitTest_1_2 instead.

File: searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp [hunk 6]
```diff
@@ -852,54 +857,58 @@ struct MaxOutstandingMoveOpsFixture : public ControllerFixture
...
     }
     void unblockJob(uint32_t expRunnerCnt) {
         _moveHandler.clearMoveDoneContexts(); // unblocks job and try to execute it via runner
-        EXPECT_EQUAL(expRunnerCnt, _runner.runCount);
+        EXPECT_EQ(expRunnerCnt, _runner.runCount);
         EXPECT_FALSE(_bmj.isBlocked());
     }
+};
 
+struct MaxOutstandingMoveOpsFixture_1 : public MaxOutstandingMoveOpsFixture {
```
Comment (geirst) [inline searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp:870]: Consider MaxOutstandingMoveOpsTest_1 instead.

File: searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp [hunk 7]
```diff
@@ -852,54 +857,58 @@ struct MaxOutstandingMoveOpsFixture : public ControllerFixture
...
+    MaxOutstandingMoveOpsFixture_1() : MaxOutstandingMoveOpsFixture(1) {}
 };
 
-TEST_F("require that bucket move job is blocked if it has too many outstanding move operations (max=1)", MaxOutstandingMoveOpsFixture(1))
+struct MaxOutstandingMoveOpsFixture_2 : public MaxOutstandingMoveOpsFixture {
```
Comment (geirst) [inline searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp:874]: Consider MaxOutstandingMoveOpsTest_2 instead.

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**